### PR TITLE
[13.0][ADD] product_variant_change_attribute_value

### PR DESCRIPTION
**Pull request by simahawk**
Replaces https://github.com/OCA/product-variant/pull/216

I'll squash some commits after approval.

### CODE REVIEW

File: product_variant_change_attribute_value/wizards/product_variant_attribute_value_action.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+            WHERE
+                attribute_id IN %(ids)s
+            GROUP BY
+                attribute_id
+        """
+        ids = tuple(self.mapped("attribute_id").ids)
+        if not ids:
+            self.update({"selectable_attribute_value_ids": False})
+            return
+        self.env.cr.execute(query, dict(ids=ids))
```
Comment (grindtildeath) [inline product_variant_change_attribute_value/wizards/product_variant_attribute_value_action.py:62]:
```suggestion
        self.env["product.attribute.value"].flush(["attribute_id"])
        self.env.cr.execute(query, dict(ids=ids))
```

File: product_variant_change_attribute_value/wizards/product_variant_attribute_value_wizard.py [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+
+import psycopg2
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class VariantAttributeValueWizard(models.TransientModel):
+    _name = "variant.attribute.value.wizard"
+    _description = "Wizard to change attriubtes on product variants"
```
Comment (grindtildeath) [inline product_variant_change_attribute_value/wizards/product_variant_attribute_value_wizard.py]:
```suggestion
    _description = "Wizard to change attributes on product variants"
```

**Review by grindtildeath — COMMENTED** [review]
I couldn't test on runbot yet but LG

**Review by grindtildeath — COMMENTED** [review]
After testing on runbot, it looks like it's quite easy to mess things up by deleting attribute values, and we're not able to recreate these on an existing product.
I'd suggest to state limitations in the readme, and restrict access to the wizard to a dedicated user group.

**Review by TDu — APPROVED** [review]

**Review by mmequignon — CHANGES_REQUESTED** [review]
https://user-images.githubusercontent.com/13200247/143858116-c6e984e1-c35a-4fc5-a1cf-2bdf116fcf59.mp4

There's a nasty bug here.

File: product_variant_change_attribute_value/tests/test_product_variant_change_attribute_value.py [hunk 3]
```diff
@@ -88,9 +89,10 @@ def test_fields(self):
     def test_actions_field_filter(self):
         wiz = self._get_wiz()
         self.assertEqual(len(wiz.attributes_action_ids), len(wiz.attribute_value_ids))
-        wiz.filter_attribute_id = self.legs
+        with Form(wiz) as res:
+            res.filter_attribute_id = self.legs
         self.assertEqual(
-            len(wiz.attributes_action_ids),
+            len(res.attributes_action_ids),
```
Comment (TDu) [inline product_variant_change_attribute_value/tests/test_product_variant_change_attribute_value.py:95]: Should the assert not be in the context manager ?

Comment (mmequignon) [reply product_variant_change_attribute_value/tests/test_product_variant_change_attribute_value.py:95]: Maybe. Probably. I have to switch on something else. I'll come back to this later :smile:

**Comment by simahawk** [comment]
@mmequignon do you think we can finalize this (maybe w/ improvements from 14)?

**Pull request closed without merging**